apps/stripe/src/app/api/saleor/payment-gateway-initialize-session/use-case.ts

```diff
@@ -51,37 +51,30 @@ export class InitializeStripeSessionUseCase {
       const pk = stripeConfigForThisChannel.value?.publishableKey;
 
       if (!pk) {
-        return err(
-          new UseCaseMissingConfigError("Config for channel not found", {
-            props: {
-              channelId,
-            },
-          }),
-        );
-      }
-
-      const rawShape: PaymentGatewayInitializeResponseShapeType = {
-        stripePublishableKey: pk.keyValue,
-      };
-
-      const responseDataShape = PaymentGatewayInitializeResponseShape.parse(rawShape);
-
-      const validResponseShape =
-        buildSyncWebhookResponsePayload<"PAYMENT_GATEWAY_INITIALIZE_SESSION">({
-          data: responseDataShape,
+        this.logger.warn("Config for channel not found", {
+          channelId,
         });
 
-      return ok(validResponseShape);
-    }
+        return err(new AppIsNotConfiguredResponse());
+      }
 
-    if (stripeConfigForThisChannel.isErr()) {
-      return err(
-        new UseCaseGetConfigError("Failed to retrieve Publishable Key from config", {
-          cause: stripeConfigForThisChannel.error,
+      return ok(
+        new PaymentGatewayInitializeSessionUseCaseResponses.Success({
+          responseData: responseData.parse({
+            stripePublishableKey: pk.keyValue,
+          }),
```

Comment:
IMO this is too early to unpack pk.

IMO
1. You should pass directly `pk` to `Success()` constructor
2. `Success()` will internally know how to construct response

This will give us actual abstraction on response shape - now you have class that returns it, but it doesn't create it. There is no reason to have abstraction of response if you pass entire response in the constrcturo from the ourside